src/Concordium/Client/GRPC.hs

```diff
@@ -343,203 +240,6 @@ getCryptographicParameters :: MonadIO m => Text -> ClientMonad m (GRPCResult Val
 getCryptographicParameters hash = withUnary (call @"getCryptographicParameters") msg getJSON
   where msg = defMessage & CF.blockHash .~ hash
 
--- |Cookie headers that may be returned by the node in a query.
-type CookieHeaders = Map.Map BS8.ByteString BS8.ByteString
```

Comment:
I am really not a fan of making these moves as part of a bigger PR since the rest of the module is staying where it was.

This lock could have stayed where it was, and imported from the other module.

Alternatively we could move the lock into a separate module as part of its own PR that does just that. And that would be merged to main directly quickly, which would make it easier to review as well.

Reply:
I agree that this should probably be in a separate PR. As I recall I had a problem when introducing rewriting the `TransactionStatusQuery` instance using the V2 API as it introduced a cyclic dependency. I'll keep this in mind from now on. Is this OK as-is for now?

Reply:
Otherwise just let me know and I will make a PR 👍🏼

Reply:
I would prefer if you make a PR moving the lock to its own module (and exporting only the required functions from there).